File: problems/Longest Substring Without Repeating Characters/lengthOfLongestSubstring.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var store = new HashSet<char>();
        var left = 0;
        var right = 0;
        var result = 0;

        while (s.Length > right) {
            if (store.Contains(s[right])) {
                store.Remove(s[left++]);
            } else {
                store.Add(s[right++]);
                result = Math.Max(result, store.Count);
            }
        }

        return result;
    }
}

File: problems/Longest Substring Without Repeating Characters/lengthOfLongestSubstring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int result = 0;
        int n = s.length();
        int left = 0;
        int right = 0;
        HashSet<Character> store = new HashSet<>();

        while (n > left) {
            while (store.contains(s.charAt(left))) {
                store.remove(s.charAt(right++));
            }
            store.add(s.charAt(left++));
            result = Math.max(result, store.size());
        }

        return result;
    }
}
